src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts

```diff
@@ -141,6 +122,12 @@ export class WorkspaceUiSettingsClientWrapper {
 
         return configObject as SavedObjectsUpdateResponse<T>;
       }
+      // for the workspace scope settings
+      // if the requestWorkspaceId is not defined
+      // then we need to skip updating workspace level setting by manually removing the placeholder
+      if (id.startsWith(CURRENT_WORKSPACE_PLACEHOLDER)) {
+        id = id.replace(`${CURRENT_WORKSPACE_PLACEHOLDER}_`, '');
```

Comment:
Though `id` is a copy, it is not a good practice to mutate function arguments.

Reply:
Shall we through error here? It does not make sense that there will be a request trying to update workspace level setting but without workspace id context.